zetaclient/evm/evm_rpc_fallback.go

```diff
@@ -0,0 +1,398 @@
+package evm
+
+import (
+	"context"
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/onrik/ethrpc"
+	"github.com/rs/zerolog"
+	"github.com/zeta-chain/zetacore/zetaclient/common"
+	"github.com/zeta-chain/zetacore/zetaclient/config"
+	"github.com/zeta-chain/zetacore/zetaclient/interfaces"
+)
+
+var _ interfaces.EthClientFallback = &EthClientFallback{}
+
+// EthClientFallback is a decorator combining client interfaces used by evm chains. Also encapsulates list of clients
+// defined by endpoints from the config.
+type EthClientFallback struct {
+	evmCfg         config.EVMConfig
+	ethClients     *common.ClientQueue
+	jsonRPCClients *common.ClientQueue
+	logger         zerolog.Logger
+}
+
+// NewEthClientFallback creates new instance of eth client used by evm chain client.
+func NewEthClientFallback(evmCfg config.EVMConfig, logger zerolog.Logger) (*EthClientFallback, error) {
+	if len(evmCfg.Endpoints) == 0 {
+		return nil, errors.New("invalid endpoints")
+	}
+	ethClientFallback := EthClientFallback{}
+	ethClientFallback.ethClients = common.NewClientQueue()
+	ethClientFallback.jsonRPCClients = common.NewClientQueue()
+
+	// Initialize clients
+	for _, endpoint := range evmCfg.Endpoints {
+		//Initialize go-ethereum clients
+		client, err := ethclient.Dial(endpoint)
+		if err != nil {
+			logger.Error().Err(err).Msg("eth Client Dial")
```

Comment:
I'm think a scenario when an observer runner provides 2 endpoints provided in the EVM config and 1/2 endpoints is not working due to some reason. In order to run/restart `zetaclient`, the operator has to manually identify and remove the glitchy endpoint from EVM config to avoid the error returned by `NewEthClientFallback`. Would it be a good idea to let `NewEthClientFallback` 

- check each endpoint in the list
- use those good ones to initialize `EthClientFallback` instance
- print an error message for identified bad endpoints
- if all endpoints are bad, return error.